httpbp/client_middlewares.go

```diff
@@ -349,3 +354,37 @@ func PrometheusClientMetrics(serverSlug string) ClientMiddleware {
 		})
 	}
 }
+
+func (c clientFaultMiddleware) Middleware() ClientMiddleware {
```

Comment:
Can we add method comments to match the amount of documentation to other middlewares, though maybe with the bulk of the information on the `NewClientFaultMiddleware` constructor? Since you've already provided great detail in the description:

> If clients are currently passing matching X-Bp-Fault headers around and they happen to be valid according to this change, then unintended fault injection could take place. This is extremely unlikely to occur given how specific the headers and their valid values are, but I wanted to call it out as technically it could affect requests unintentionally.

Reply:
done!